app/support/order_row_importer.rb

```diff
@@ -70,10 +70,14 @@ def order_key
     @order_key ||= [user_field, chart_string_field, order_date_field]
   end
 
-  def row_with_errors # TODO: refactor
-    new_row = @row.dup
+  def row_with_errors
+    # Start with a hash of HEADERS keys with nil values to ensure optional columns
+    # are included in the report even if they are not in the uploaded CSV.
```

Comment:
No, it's actually nil values. If you zip and one is longer than the other you get nils.

`HEADERS.values.zip []` => [["Header 1", nil], ["Header 2", nil], ...]`, then `Hash[]` => `{ "Header 1" => nil, "Header 2" => nil, ... }`. I'm open to better ways to get to this, or describe it. Ultimately, what I need is to make sure `new_row` has all the expected columns. Before, it might not have had "Notes", which is why the "Errors" were ending up in that column sometimes.